libs/wingsdk/src/shared-gcp/bucket.inflight.ts

```diff
@@ -0,0 +1,238 @@
+import { Storage, Bucket } from "@google-cloud/storage";
+import {
+  BucketDeleteOptions,
+  IBucketClient,
+  ObjectMetadata,
+  SignedUrlOptions,
+} from "../cloud";
+import { Json } from "../std";
+
+export class BucketClient implements IBucketClient {
+  private _public: boolean;
+  private bucketName: string;
+  private storage: Storage;
+  private bucket: Bucket;
+
+  constructor(
+    bucketName: string,
+    isPublic: boolean = false,
+    storage: Storage,
+    projectId: string = process.env.GCP_PROJECT_ID || "wingsdk-test"
+  ) {
+    this._public = isPublic;
```

Comment:
is there a way to make the bucket itself private?

Reply:
Yes there is. `this.bucket.makePrivate()` and `this.bucket.makePublic()` works well. But I don't think `IBucketClient` interface support any of related to this right now for making a bucket public or private. If it is something we need, I can definitely implement that. 

Right now I'm using this `isPublic` props just for unit testing.

Reply:
I see, 
I guess it's implemented in `target-tf-gcp/bucket.ts`, 
is there a way to query that on the inflight client?